flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/JsonFunctionsITCase.java

```diff
@@ -821,12 +821,18 @@ private static List<TestSetSpec> jsonSpec() {
                                         json("[1, 2, 3]")),
                                 "JSON_OBJECT(KEY 'p1' VALUE JSON(f0), KEY 'p2' VALUE JSON(f1), KEY 'p3' VALUE JSON('[1, 2, 3]'))",
                                 "{\"p1\":{\"key\":\"value\"},\"p2\":{\"key\":{\"value\":42}},\"p3\":[1,2,3]}",
-                                STRING().notNull()),
-
-                // Tests for JSON calls inside of JSON_ARRAY
-                TestSetSpec.forFunction(BuiltInFunctionDefinitions.JSON_ARRAY)
-                        .onFieldsWithData("{\"key\":\"value\"}", "{\"key\": {\"value\": 42}}")
-                        .andDataTypes(STRING(), STRING())
+                                STRING().notNull())
+                        .testSqlValidationError(
+                                "JSON_OBJECT(KEY JSON('{}') VALUE 'value' ABSENT ON NULL)",
+                                "The JSON() function is currently only supported inside a JSON_OBJECT() or JSON_ARRAY() function.")
```

Comment:
The error message is confusing though. `JSON()` function is indeed inside `JSON_OBJECT`. We need to make it more accurate that it also needs to be in value position?

Reply:
The full message contains examples that should help the user understand the expected use. [See](https://github.com/apache/flink/blob/e30c207e371794adaac0261ed20a58a7c812f192/flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/ExprCodeGenerator.scala#L467):

```
The JSON() function is currently only supported inside a JSON_OBJECT() or JSON_ARRAY()" +
          " function. Example: JSON_OBJECT('a', JSON('{"key": "value"}')) or " +
          "JSON_ARRAY(JSON('{"key": "value"}')).
```

Do you think that's clear enough? Let me know if you have a specific suggestion

Reply:
`The JSON() function is currently only supported inside a JSON_OBJECT() or JSON_ARRAY() as values`? Is this more accurate?

Reply:
The param is actually called `VALUE`. What do you think:

`The JSON() function is currently only supported inside JSON_ARRAY() or as the VALUE param of JSON_OBJECT()`